core/src/main/java/com/amplifyframework/analytics/BooleanProperty.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.analytics;
+
+import androidx.annotation.NonNull;
+
+/**
+ * BooleanProperty wraps a Boolean value to store in {@link Properties}.
+ */
+public final class BooleanProperty implements Property<Boolean> {
+    private final Boolean value;
+
+    private BooleanProperty(Boolean value) {
+        this.value = value;
+    }
+
+    /**
+     * getValue returns the wrapped Boolean value stored in the property.
+     *
+     * @return The wrapped Boolean value
+     */
+    @Override
+    public Boolean getValue() {
+        return value;
+    }
+
+    /**
+     * Factory method to instantiate a {@link BooleanProperty} from a {@link Boolean} value.
+     *
+     * @param value a boolean value
+     * @return an instance of {@link BooleanProperty}
+     */
+    public static BooleanProperty from(@NonNull Boolean value) {
+        return new BooleanProperty(value);
```

Comment:
Potentially, `BooleanProperty(Objects.requireNonNull(value))`  (Same for all of the arguments to various `FooProperty` factories.)